AdminCore.Services/EventService.cs

```diff
@@ -127,17 +129,63 @@ public void UpdateEventStatus(int eventId, EventStatuses status)
       }
     }
 
-    public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, int employeeId)
+    public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, int employeeId, int eventWorkflowId)
     {
       CheckEventTypeAdminLevel(eventTypes, employeeId);
-      var newEvent = BuildNewEvent(employeeId, eventTypes);
+      var newEvent = BuildNewEvent(employeeId, eventTypes, eventWorkflowId);
       UpdateEventDates(dates, newEvent);
+      ThrowIfDaysNoticeValidationFail((int)eventTypes, newEvent.EventDates);
       return ValidateRemainingHolidaysAndCreate(newEvent, dates);
     }
 
-    public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, Employee employee)
+    /// <summary>
+    /// Throws exception if the leave is requested outside of the required notice period.
+    /// Notice period definitions stored in database.
+    /// Example:
+    /// Today: 20/01/2000, LeaveStart: 21/01/2000, LeaveEnd: 21/01/2000
+    /// leaveLengthDays = 1
+    /// Doesn't take into account weekends.
+    /// </summary>
+    /// <param name="eventTypeId"></param>
+    /// <param name="eventDates"></param>
+    /// <exception cref="ValidationException"></exception>
+    private void ThrowIfDaysNoticeValidationFail(int eventTypeId, ICollection<EventDate> eventDates)
     {
-      var newEvent = BuildNewEvent(employee, eventTypes);
+      var currDate = _dateService.GetCurrentDateTime();
+
+      int leaveLengthDays = 0;
```

Comment:
this could be a LINQ expression
```
var leaveLengthDays = eventDates.Sum(eventDate => eventDate.StartDate.BusinessDaysUntil(eventDate.EndDate));
```

Reply:
Done